projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { Observable, of } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+import { LocationModel } from '../../models/location.model';
+import { LocationService } from '../../services/locations/location.service';
+
+@Component({
+  selector: 'exui-search-location',
+  templateUrl: './search-location.component.html',
+  styleUrls: ['./search-location.component.scss']
+})
+
+export class SearchLocationComponent implements OnInit {
+  @Input() public serviceId: string = '';
+  @Input() public locationType: string = '';
+  @Input() public disabled?: boolean = null;
+  @Input() public selectedLocations$: Observable<LocationModel[]>;
+  @Input() public submitted?: boolean = true;
+  @Input() public findLocationForm: FormGroup;
+
+  public locations$: Observable<LocationModel[]>;
+  public selectedLocation: LocationModel;
+  public showAutocomplete = false;
+
+  private readonly minSearchCharacters = 2;
+
+  constructor(private readonly locationService: LocationService, fb: FormBuilder) {
+    this.findLocationForm =  fb.group({
+      findLocationControl: [null],
+    });
+
+    this.selectedLocations$ = of([]);
+  }
+
+  public ngOnInit(): void {
+    this.locations$ = this.getLocations('');
+
+    this.findLocationForm.controls.findLocationControl.valueChanges.subscribe(value => {
+      this.search(value);
+    });
+  }
+
+  public filter(term: string) {
+     this.getLocations(term).pipe(
+        mergeMap((apiData: LocationModel[]) => this.selectedLocations$.pipe(
+        map((selectedLocations) => apiData.filter(apiLocation => !selectedLocations.map(x => x.court_venue_id).includes(apiLocation.court_venue_id)))
```

Comment:
could this x change to something meaningful?

Reply:
Please elaborate all functions and names named appropriately.

Reply:
Oh ok I understand

Reply:
selectedLocation